src/Factory/HalViewHelperFactory.php

```diff
@@ -30,7 +30,8 @@ public function createService(ServiceLocatorInterface $serviceLocator)
         }
         $urlHelper       = $serviceLocator->get('Url');
 
-        $helper = new Plugin\Hal($hydrators);
+        /** @var Plugin\Hal $helper */
+        $helper = $services->get('ControllerPluginManager')->get('Hal');
```

Comment:
This is the problematic change -- because the ControllerPluginManager currently pulls the Hal view helper from the ViewHelperManager and returns it. The test passes because you mock the ControllerPluginManager and what it returns when the Hal plugin is requested.

Reverting this change will fix the issue. I'll get a PR in place for #81 shortly.